src/components/screens/consul/Proposals/NewProposal.js

```diff
@@ -270,7 +318,92 @@ export const NewProposal = ({ navigation, data, query }) => {
   );
 };
 
+const ImageSelector = ({ control, field, item, selectImage }) => {
+  const [errorMessage, setErrorMessage] = useState('');
+  const [imageInfoText, setImageInfoText] = useState('');
+
+  const { buttonTitle, infoText } = item;
+  const { name, onChange, value } = field;
+
+  return (
+    <>
+      <Input
+        {...item}
+        control={control}
+        errorMessage={errorMessage}
+        hidden
+        validate
+        name={name}
+        value={value}
+      />
+      <RegularText smallest placeholder>
+        {infoText}
+      </RegularText>
+
+      {value ? (
+        <>
+          <WrapperRow center spaceBetween>
+            <Image source={{ uri: value }} style={styles.image} />
+
+            <TouchableOpacity
+              onPress={() => {
+                onChange('');
+                setErrorMessage('');
+              }}
+            >
+              <Icon.Trash color={colors.error} size={normalize(16)} />
+            </TouchableOpacity>
+          </WrapperRow>
+          <RegularText smallest>{imageInfoText}</RegularText>
+        </>
+      ) : (
+        <Button
+          title={buttonTitle}
+          invert
+          onPress={async () => {
+            const { uri, type } = await selectImage();
+            const { size } = await FileSystem.getInfoAsync(uri);
+            const errorMessage = imageErrorMessageGenerator({
+              size,
+              uri: uri.substr(uri.length - 4)
+            });
+
+            setErrorMessage(texts.consul.startNew[errorMessage]);
+            setImageInfoText(`(${type}/${uri.substr(uri.length - 3)}, ${bytesToSize(size)})`);
+            onChange(uri);
+          }}
+        />
+      )}
+    </>
+  );
+};
+
+const bytesToSize = (bytes) => {
+  const sizes = ['Bytes', 'KB', 'MB', 'GB', 'TB'];
+  if (bytes == 0) return '0 Byte';
+  let i = parseInt(Math.floor(Math.log(bytes) / Math.log(1024)));
+
+  return Math.round(bytes / Math.pow(1024, i), 2) + ' ' + sizes[i];
+};
+
+const imageErrorMessageGenerator = ({ size, uri }) => {
+  const errorMessage =
+    uri !== '.jpg' && uri !== 'jpeg' && size > 1048576
+      ? 'choose-image-content-type-image/png-does-not-match-any-of-accepted-content-types-jpg,-choose-image-must-be-in-between-0-bytes-and-1-mb'
+      : uri !== '.jpg' && uri !== 'jpeg'
+      ? 'choose-image-content-type-image/png-does-not-match-any-of-accepted-content-types-jpg'
+      : size > 1048576
+      ? 'choose-image-must-be-in-between-0-bytes-and-1-mb'
+      : '';
```

Comment:
why `.jpg` but `jpeg` without `.`? ;)

please use consts for `.jpg`, `.jpeg` and `1048576`  inside of this function to not duplicate these values.

example:

```
const isJPG = uri === '.jpg' || uri === '.jpeg';
const isGreater1MB = size > 1048576;
```

then you can use `errorMessage = !isJPG && !isGreater1MB ? ...` and so on, which is more readable.

the `uri` is always just the file ending/file type and not the whole path?

Reply:
Yes, I didn't think about it that way. Thank you. 

While sending the Uri data to the function, it is shown as type in the function because I clip it.

Reply:
ok, so when its clipped, it is not a `uri` anymore. this needs to be named as something like `imageType` please